### PR TITLE
OSD (WIP)

### PR DESCRIPTION
**Pull request by btzr-io**
## Todo 
- [x] Pinch to Zoom https://github.com/workhorsy/comic_book_reader/issues/85
- [x] Fix drag / drop.
- [x] Upload / open files.
- [x] Connect `unistore` ( actions / reducers)
- [x] Embed api. https://github.com/workhorsy/comic_book_reader/issues/118 :dart:
- [x] More fixes for redesign.
- [x] OpenSeaDragon integration :dart: 
- [ ] Connect settings.
- [x] Open actual comic files :dart: 
- [x] Book-mode (display two pages)
- [ ] Handle errors.

https://openseadragon.github.io/docs/OpenSeadragon.Viewer.html#.event:open-failed
https://openseadragon.github.io/docs/OpenSeadragon.Viewer.html#.event:add-item-failed

### CODE REVIEW

File: src/routes/viewer/components/reader/index.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  componentDidMount() {
+    // Initializa OSD instance
+    this.initOpenSeaDragon()
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // Quick hack: (TODO) FIX IT!
+    const { bookMode, currentPage } = this.props.reader
+    const nextBookMode = nextProps.reader.bookMode
+    if (bookMode != nextBookMode) {
```
Comment (btzr-io) [inline src/routes/viewer/components/reader/index.js:154]: Not sure about this, needs a better way to handle this :confused:

**Comment by btzr-io** [comment]
:dart:  Performance should be a priority, I need to reduce component updates :stuck_out_tongue:

**Comment by workhorsy** [comment]
I just tried it out again. Loads comics and displays them in OSD. Good job! It is coming along nicely.

**Comment by workhorsy** [comment]
Do you want me to merge this?

**Comment by btzr-io** [comment]
@workhorsy I'm still working on this, I'll let you know when it's ready :v:

**Comment by btzr-io** [comment]
@workhorsy I guess now you can merge this :+1:

**Pull request merged**